pennylane/operation.py

```diff
@@ -534,6 +535,13 @@ def parameters(self):
         """Current parameter values."""
         return self.data.copy()
 
+    @property
+    def hyperparameters(self):
+        """Non-trainable variables that define this operation."""
+        if hasattr(self, "_hyperparameters"):
+            return self._hyperparameters
+        return {}
```

Comment:
```suggestion
        self._hyperparameters = {}
        return self._hyperparameters
```

If someone is trying to set a key using dictionary mutability (which I think should be the way), the previous logic would give you the false sense that you succeeded.

But if `_hyperparameters` is created in `__init__`, I think we can always assume it exists. And not check with `hasattr`.

Reply:
Sounds good